flake.nix

```diff
@@ -0,0 +1,214 @@
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ self.overlay ]; });
+    in {
+      overlay = final: prev:
+        with final; {
+          kiwiirc = mkYarnPackage rec {
+            src = ./.;
+            pname = "kiwiirc";
+            distPhase = "true";
+            buildPhase = ''
+              yarn --offline run build
+            '';
+            preInstall = ''
+              mkdir -p $out/www/${pname}
+              cp -r ./deps/${pname}/dist/* $out/www
+              cp -r $src/LICENSE $out/www
+            '';
+            fixupPhase = ''
+              rm -rf $out/tarballs $out/libexec $out/bin
+              rm -rf $out/www/kiwiirc
+            '';
+          };
+          kiwiirc-desktop = let executableName = "kiwiirc-desktop";
+          in mkYarnPackage rec {
+            name = "kiwiirc-desktop";
+            src = kiwiirc-desktop;
+            patches = ((writeText "remove-dev-mode.patch" ''
+              diff --git a/src/index.js b/src/index.js
+              index 1ac9c27..d5cbb79 100644
+              --- a/src/index.js
+              +++ b/src/index.js
+              @@ -141,9 +141,4 @@ app.on('activate', async () => {
+                           app.quit();
+                       });
+                   }
+              -
+              -    if (process.defaultApp) {
+              -        // Running in dev mode
+              -        mainWindow.webContents.openDevTools();
+              -    }
+               })();
+            ''));
+            nativeBuildInputs = [ makeWrapper ];
+            installPhase = ''
+              # resources
+              mkdir -p "$out/share/kiwiirc"
+              cp -r './deps/kiwiirc-desktop' "$out/share/kiwiirc/electron"
+
+              rm "$out/share/kiwiirc/electron/node_modules"
+              cp -r './node_modules' "$out/share/kiwiirc/electron"
+
+              rm -r "$out/share/kiwiirc/electron/kiwiirc"
+              mkdir -p  "$out/share/kiwiirc/electron/kiwiirc"
+              ln -s '${final.kiwiirc}/www' "$out/share/kiwiirc/electron/kiwiirc/dist"
+
+              # icons
+              for res in 128x128 256x256 512x512; do
+                mkdir -p "$out/share/icons/hicolor/$res/apps"
+                cp "./deps/kiwiirc-desktop/static/icons/kiwiirclogo_$res.png" "$out/share/icons/hicolor/$res/apps/kiwiirc.png"
+              done
+
+              # desktop item
+              mkdir -p "$out/share"
+              ln -s "${desktopItem}/share/applications" "$out/share/applications"
+
+              # executable wrapper
+              makeWrapper '${electron}/bin/electron' "$out/bin/${executableName}" \
+                --add-flags "$out/share/kiwiirc/electron"
+            '';
+
+            distPhase = ''
```

Comment:
With `mkYarnPackage` you can't set `dontDist = true;`, because if you do, it will override it. So, to get around that fact, you set `distPhase` to an empty string. This way, it effectively disables the `distPhase` for `mkYarnPackage`.